artemis-server/src/main/java/org/apache/activemq/artemis/core/server/MirrorOption.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.server;
+
+/** This is to be used in conjunction with RoutingContext, where we control certain semantics during routing.
+ *  */
+public enum MirrorOption {
```

Comment:
this is for the Mirror option , there's an interface in the broker... it's either disable... I don't think we can give another name.

Reply:
I only see this being used to set a toggle on RoutingContext? A 'MirrorOption' enum sitting at the root of the server package tree, would to me most typicaly suggest its about setting options for the Mirror itself. This isnt, its about controlling RoutingContext behaviour. MirrorOption is perhaps the last name I would give it from how I see it being used.

Reply:
Even just nesting it in RoutingContext would seem nicer. RoutingContext.MirrorControl{enabled, disabled, individual}?

Reply:
I had it nestes at first. But then the syntax got better in other usages. 

The way I see this is, it is controlling how Mirror happens after routing.

Reply:
I prefer it as a separate class to be honest.

the name Control to me implies it's doing control.. in this case it's just really an option on what is the outcome of Mirroring at the current Routing.

Reply:
Having it occupy the 'prime real estate' of the name MirrorOption at the very root of the code namepsace seems wrong given what its used for. Its not for holding general options for the mirror, its a fairly niche almost-internal routing control setting for use with RoutingContext.

Even if were left it named MirrorOption, it still seems like it would be far nicer as a nested enum within RoutingContext, the only place it is used and seems likely to be used. None of the impl-only code using it would change really beyond just updating their import (probably even removing one for RoutingContextImpl actually)